__tests__/commands/install/integration.js

```diff
@@ -764,18 +764,18 @@ if (process.platform !== 'darwin') {
     });
 }
 
-// Covers current behavior, issue opened whether this should be changed https://github.com/yarnpkg/yarn/issues/2274
-test.concurrent('optional dependency that fails to build should still be installed',
+test.concurrent('optional dependency that fails to build should not be installed',
   (): Promise<void> => {
-    return runInstall({}, 'should-install-failing-optional-deps', async (config) => {
-      assert.ok(await fs.exists(path.join(config.cwd, 'node_modules', 'optional-failing')));
+    return runInstallWithoutCheck({}, 'should-not-install-failing-optional-deps', (config) => {
+      assert.equal(fsNode.existsSync(path.join(config.cwd, 'node_modules', 'optional-failing')), false);
```

Comment:
Thanks, I did not know that.
We'll probably do a one-off codemod to change to fs.access everywhere